src/models/thread.ts

```diff
@@ -624,6 +626,8 @@ export class Thread extends ReadReceipt<ThreadEmittedEvents, ThreadEventHandlerM
             }
         }
 
+        this.checkForMissingReceiptEvent();
```

Comment:
is it deliberate that this lacks an `await`? If so, might be good to comment it to highlight and explain why.

Reply:
It wasn't, but it now has the throttle where the wrapper doesn't return a promise.